src/meepgeom.cpp

```diff
@@ -29,13 +29,13 @@ material_data vacuum_material_data;
 material_type vacuum = &vacuum_material_data;
 
 static void set_default_material(material_type _default_material) {
-  if (default_material != NULL) {
+  if ((default_material != NULL) && (default_material != _default_material)) {
```

Comment:
It might be simpler to have an early exit like this ?

if (default_material == _default_material) return;

and then avoid those checks later ?

Reply:
Good suggestion. Fixed.